pkg/controller/datavolume-controller.go

```diff
@@ -595,24 +546,201 @@ func (r *DatavolumeReconciler) reconcileSmartClonePvc(log logr.Logger, datavolum
 				return reconcile.Result{}, err
 			}
 		} else if tmpPVC.Annotations[AnnCloneOf] == "true" {
-			done, err := r.expand(log, datavolume, tmpPVC, pvcSpec)
-			if err != nil {
-				return reconcile.Result{}, err
-			}
-			if !done {
-				return reconcile.Result{},
-					r.updateSmartCloneStatusPhase(cdiv1.ExpansionInProgress, datavolume, nil)
-			}
+			return r.expandTmpPvcAfterClone(log, datavolume, pvcSpec, tmpPVC, CsiClone)
+		}
+	}
+
+	if datavolume.Status.Phase == cdiv1.NamespaceTransferInProgress {
+		return reconcile.Result{}, nil
+	}
+
+	sourcePvcNs := datavolume.Spec.Source.PVC.Namespace
+	if sourcePvcNs == "" {
+		sourcePvcNs = datavolume.Namespace
+	}
+	r.log.V(3).Info("CSI-Clone is available")
+
+	// Get source pvc
+	sourcePvc := &corev1.PersistentVolumeClaim{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: sourcePvcNs, Name: datavolume.Spec.Source.PVC.Name}, sourcePvc); err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.V(3).Info("Source PVC no longer exists")
+			return reconcile.Result{}, err // r.updateCSICloneDataVolumeStatus(cdiv1.CloneSourcePVCLost, datavolume)
+		}
+		return reconcile.Result{}, err
+	}
+
+	inUse, err := r.sourceInUse(datavolume, CSICloneSourceInUse)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	populated, err := r.isSourcePVCPopulated(datavolume)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if inUse || !populated {
+		return reconcile.Result{Requeue: true},
```

Comment:
This is an okay use of requeue because we may not be watching/reconciling these pods